### PR TITLE
Fixed bugs with UnionType and Array (when defining items as a list).

### PR DESCRIPTION
**Pull request by boralyl**
Also added items type(s) documentation for arrays in api documentation.  This PR address #108 and #109 .

### CODE REVIEW

File: doctor/docs/base.py [hunk 1]
```diff
@@ -121,10 +121,101 @@ def get_example_lines(headers: Dict[str, str], method: str, url: str,
     return lines
 
 
+def get_object_reference(obj: Object) -> str:
```
Comment (boralyl) [inline doctor/docs/base.py:124]: I just made this it's own function as we duplicated this logic in 3 or 4 places.

File: doctor/docs/base.py [hunk 2]
```diff
@@ -121,10 +121,101 @@ def get_example_lines(headers: Dict[str, str], method: str, url: str,
...
+    """
+    resource_name = obj.title
+    if resource_name is None:
+        class_name = obj.__name__
+        resource_name = class_name_to_resource_name(class_name)
+    ALL_RESOURCES[resource_name] = obj
+    return ' See :ref:`resource-{}`.'.format(
+        '-'.join(resource_name.split(' ')).lower().strip())
+
+
```
Comment (boralyl) [inline doctor/docs/base.py:141]: New function to return documentation about the type(s) of an array's items.

File: doctor/docs/base.py [hunk 3]
```diff
@@ -121,10 +121,101 @@ def get_example_lines(headers: Dict[str, str], method: str, url: str,
...
+        if issubclass(item.items, Enum):
+            _enum = ' Must be one of: `{}`.'.format(
+                item.items.enum)
+        elif issubclass(item.items, Object):
+            ref = get_object_reference(item)
+        desc = '  *Items must be*: {description}{enum}{ref}'.format(
+            description=desc, enum=_enum, ref=ref)
+    return desc
+
+
```
Comment (boralyl) [inline doctor/docs/base.py:181]: Made this it's own function to clean up the code a bit as it was making `get_json_object_lines` a little long when I added the array specific logic.

File: doctor/docs/base.py [hunk 4]
```diff
@@ -149,34 +240,40 @@ def get_json_object_lines(annotation: ResourceAnnotation,
...
         obj_ref = ''
         if issubclass(annotated_type, Object):
-            resource_name = annotated_type.title
-            if resource_name is None:
-                class_name = annotated_type.__name__
-                resource_name = class_name_to_resource_name(class_name)
-            ALL_RESOURCES[resource_name] = annotated_type
-            obj_ref = ' See :ref:`resource-{}`.'.format(
-                '-'.join(resource_name.split(' ')).lower().strip())
-        elif (hasattr(annotated_type, 'items') and
```
Comment (boralyl) [inline doctor/docs/base.py:163]: I had to change this to `issubclass(annotated_type, Array)` as Object types subclass dict which has an `items` attr.

File: doctor/types.py [hunk 5]
```diff
@@ -116,6 +139,15 @@ def __new__(cls, *args, **kwargs):
 
         return value
```
Comment (boralyl) [inline doctor/types.py:141]: Fixes #108

File: doctor/types.py [hunk 6]
```diff
@@ -534,7 +566,10 @@ def get_example(cls) -> list:
         if cls.example is not None:
             return cls.example
         if cls.items is not None:
```
Comment (boralyl) [inline doctor/types.py:568]: Fixes #109

**Review by pkar — APPROVED** [review]

**Pull request merged**